src/grammar/grammar.js

```diff
@@ -176,11 +176,33 @@ export default class Grammar {
    * Reads grammar file data. Supports reading `bnf`,
    * and `lex` grammars based on mode.
    */
-  static dataFromGrammarFile(grammarFile, grammarType = 'bnf') {
-    return Grammar.dataFromString(
-      fs.readFileSync(grammarFile, 'utf-8'),
-      grammarType
-    );
+  static dataFromGrammarFile(grammarFile, grammarType = 'bnf', loc = false) {
```

Comment:
Let's use options here (potentially can refactor `grammarType` into the options as well):

```js
static dataFromGrammarFile(grammarFile, {grammarType = 'bnf', useLocation = false}) {
  ...
}
```